components/raftstore/src/coprocessor/dispatcher.rs

```diff
@@ -202,7 +202,8 @@ macro_rules! push {
         };
         let vec = &mut $vec;
         vec.push(e);
-        vec.sort_by(|l, r| l.priority.cmp(&r.priority));
+        // reverse sorting, so observer with higher priority will call first
+        vec.sort_by(|l, r| r.priority.cmp(&l.priority));
```

Comment:
/cc @BusyJay

Reply:
I don't think it's a good idea, low value means higher priority is a common sense. For example, linux considers lower nice value as higher priority; Java considers least element is on the head of PriorityQueue.

Reply:
So this is a bug?
https://github.com/tikv/tikv/blob/70ee9c22e2c2485586d280dd0b1a39b353ffced6/components/raftstore/src/coprocessor/dispatcher.rs#L322-L342

Reply:
Yes. But I don't think `TableCheckObserver` needs to have higher priority as every key triggered by observer is a valid split key.